### PR TITLE
Redefined scope and refactored

### PR DESCRIPTION
**Pull request by ghost**
The scope has been redefined : Burrows only manages message queues / events are managed by evaneos/events.

The classes have been refactored and some  interfaces have been created

### CODE REVIEW

**Comment by maffpool** [comment]
I took a quick look. It sounds good.
By the way, it seems you removed the exchange/queue binding information. When you'll run a worker for the 1st time on a new queue, the AmqService doesn't know the exchange to bind to.
Maybe I missed something .. Do you plan to configure bindings another way ?

**Comment by ghost** [comment]
@maffpool you're right, when you declare your worker, it doesn't link the queue to the exchange. The reason why is that the worker only consumes a queue, it doesn't need to know if it comes from an exchange or if the message have been put in the queue directly. So the binding is only done when you add a queue to an exchange, or if you configure it beforehand.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 /vendor/
+
+.project
```
Comment (maffpool) [inline .gitignore:3]: Par contre, faudrait éviter de mettre ce genre de trucs dans les gitignore.
On crée des lib open source, on devrait pas mettre de code IDE-specific dedans à mon sens

Comment (maffpool) [inline .gitignore:3]: Sorry I forgot that we must write english comments 
#holidays

**Pull request merged**